### ISSUE TITLE
Support co-browser solutions such as Surfly.

### ISSUE THREAD

**Issue by yoshikiohshima** [processing/p5.js-web-editor#1866]
- New feature request

#### New feature details:

There are some co-browsing services where multiple people can open the same page at the same time, and collaborate in real time. One of such a service is called Surfly (just an example). The way such a service typically works is to rewrite code so that the  elements created are aware of remote participants. To do so, however the code has to go through the proxy that does its job. The problem with editor.p5js.org is that the `defaultHTML` string in `createDefaultFiles.js` contains hardcoded URLs to cloudflare, the proxy server cannot detect the script and fails due to the Content-Security-Policy violation.

(Being able to run p5 editor on Surfly would be really great, as a learner and a mentor can edit the same code together.)

The same issue applies to 'PREVIEW_SCRIPTS_URL', and 
It is somewhat related to the fact that the donation page https://p5js.org/download/support.html shows CSP violation errors.  Basically the server is too strict on the policy

If you create a trial account on surfly.com, you can try running editor.p5js.org. When you press the run triangle button, it fails with an error:

<img width="1003" alt="surfly" src="https://user-images.githubusercontent.com/197249/118381187-43633480-b59d-11eb-8e44-71f4691ec037.png">
. 

Basically `createDefaultFiles.js` would use the actual window.location to construct the URL to load p5.js and p5.sound.min.js, and also make 'PREVIEW_SCRIPTS_URL' dynamically adapt to the server location. (perhaps only when it sees that it is proxified.) Then the P5 editor would work in a co-browser and folks online can collaborate in real time.

**Comment by kjhollen** [comment]
Hi @yoshikiohshima, thanks for this suggestion! We are also tracking real-time collaboration in issue #1337—please feel free to add further comments there. I'm going to close this issue for now, so we can keep all of the discussion about collaboration in one place.
